core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -128,26 +128,30 @@
     public static void verifySourceAggregations(
             List<Expression> groupByExpressions,
             Scope sourceScope,
-            Expression expression,
+            List<Expression> expressions,
             Session session,
             Metadata metadata,
             Analysis analysis)
     {
         AggregationAnalyzer analyzer = new AggregationAnalyzer(groupByExpressions, sourceScope, Optional.empty(), session, metadata, analysis);
-        analyzer.analyze(expression);
+        for (Expression expression : expressions) {
+            analyzer.analyze(expression);
+        }
     }
 
     public static void verifyOrderByAggregations(
             List<Expression> groupByExpressions,
             Scope sourceScope,
             Scope orderByScope,
-            Expression expression,
+            List<Expression> expressions,
             Session session,
             Metadata metadata,
             Analysis analysis)
     {
         AggregationAnalyzer analyzer = new AggregationAnalyzer(groupByExpressions, sourceScope, Optional.of(orderByScope), session, metadata, analysis);
```

Comment:
Can `AggregationAnalyzer` be used multiple times for different expressions? It is stateful, can the previous `analyze` affect the next `analyze` call?

Reply:
From my understanding, I think it can be used multiple times for a given `groupByExpression` - the `Analyzer` which we use there just validates the output expression and it doesn't store or update any states.